### ISSUE TITLE
illegal_argument_exception while reindexing

### ISSUE THREAD

**Issue by c0da** [elastic/elasticsearch#27533]
Hello everyone,

I'm trying to migrate from 5.6.4 to 6.0, and upgrade assistant detected some indices that need to be reindexed.

While trying to reindex, some of them are failing with the following error:
```
{
  "statusCode": 500,
  "error": "Internal Server Error",
  "message": "An internal server error occurred",
  "code": "ERR_CREATE_INDEX_FAILED",
  "root_cause": [
    {
      "type": "illegal_argument_exception",
      "reason": "unknown setting [index.archived.index.routing.allocation.disable_allocation] please check that any required plugins are installed, or check the breaking changes documentation for removed settings"
    }
  ],
  "type": "illegal_argument_exception",
  "reason": "unknown setting [index.archived.index.routing.allocation.disable_allocation] please check that any required plugins are installed, or check the breaking changes documentation for removed settings",
  "reindexedIndexName": "exchange-2016.05.26-reindexed-v5"
}
```
Does anybody know to fix this?

Thanks in advance!

**Comment by albertzaharovits** [comment]
Hello @c0da 

The index you are trying to migrate to v6 has been migrated in the past from v2 to v5.
Unfortunately you have hit a double corner case:
1. the automatic migrate tool does not handle it (and causes the error you reported)
2. a bug preventing the removal of the `archived` index settings namespace https://github.com/elastic/elasticsearch/issues/27537

The workaround is to recreate the indices and reindex manually:
1. identify indices requiring upgrade with: [GET /_xpack/migration/assistance](https://www.elastic.co/guide/en/elasticsearch/reference/6.0/migration-api-assistance.html)
2. [get index settings](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-settings.html) and remove the archived namespace and any settings that have been [changed breakingly](https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking_60_indices_changes.html)
3. use the [reindex API](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html)

**Comment by c0da** [comment]
Thank you @albertzaharovits ! I'll give it a try and let you know.

**Comment by c0da** [comment]
Thank you again, @albertzaharovits ! Your workaround worked like a charm!

**Comment by tomsommer** [comment]
This is not solved by #27537 though

**Comment by andrew-goldstein** [comment]
If while migrating from v2 you run into this issue, where the `index.archived.index.routing.allocation.disable_allocation` setting prevents migration of an index via the assistant UI, you may execute the procedure documented at https://www.elastic.co/guide/en/kibana/current/migrating-6.0-index.html
for every index that contains the `index.archived.index.routing.allocation.disable_allocation` setting.

Please note that the documented procedure does not migrate existing index settings. That's generally fine for the `.kibana` index, however if you are migrating other indexes, and you wish to preserve their settings, you must do the following (for each index being migrated):

1) Copy the current index settings from the output of the following (example) `GET` request, run in the Kibana Dev Tools. If, for example, your index was named `index_test`, execute the following `GET` request:

`GET /index_test/_settings`

2) In a text editor, remove the `index.archived.index.routing.allocation.disable_allocation` setting from the output above.

3) Use the modified settings above (that omit `index.archived.index.routing.allocation.disable_allocation`) when executing the `PUT` described in step 2 of https://www.elastic.co/guide/en/kibana/current/migrating-6.0-index.html


To be clear, the extra steps above are only required if you have specific index settings you wish to retain.
